octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py

```diff
@@ -64,121 +74,279 @@ def __init__(self, config: object, exchange_manager: object):
         }
         self._set_async_callbacks()
 
-    def _set_async_callbacks(self):
-        """
-        Prevent `inspect.iscoroutinefunction` to return False when callback are cythonized
-        """
-        for callback in self.callback_by_feed.values():
-            callback.is_async = True
+        # Creates cryptofeed exchange instance
+        self.cryptofeed_exchange = cryptofeed_exchanges.EXCHANGE_MAP[self.get_feed_name()](config=self.client.config)
+
+    """
+    Abstract methods
+    """
 
     @classmethod
     def get_feed_name(cls):
         raise NotImplementedError("get_feed_name not implemented")
 
-    def fix_logger(self):
+    """
+    Abstract implementations
+    """
+
+    def start(self):
+        try:
+            self._filter_exchange_pairs_and_timeframes()
+            self._subscribe_feeds()
+        except Exception as e:
+            self.logger.exception(e, True, f"Failed to subscribe when creating websocket feed : {e}")
+        try:
+            # without this two lines `There is no current event loop in thread`
+            # is raised when calling `client.run()`
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+            self.client.run()
+        except Exception as e:
+            self.logger.error(f"Failed to start websocket feed : {e}")
+        self.logger.warning("Websocket master thread terminated, this should not happen during bot run"
+                            " with a valid configuration")
+
+    def stop(self):
+        # let the thread stop by itself
+        # self.client.stop()
+        pass
+
+    def close(self):
+        # prevent FeedHandler : "run the AsyncIO event loop one last time"
+        # self.client.close()
+        pass
+
+    def add_pairs(self, pairs):
         """
-        Replace cryptofeed feedhandler logger because it writes logs into "cryptofeed.log" file
+        Add new pairs to self.filtered_pairs
+        :param pairs: the list of pair to add
         """
-        cryptofeed.feedhandler.LOG = self.logger
-        cryptofeed.connection_handler.LOG = self.logger
+        for pair in pairs:
+            self._add_pair(self.get_exchange_pair(pair))
 
-    def fix_signal_handler(self):
+    def add_time_frames(self, time_frames):
         """
-        Websocket are started in a new thread thus signal handle cannot be used
-        add_signal_handler() can only be called from the main thread
+        Add new time_frames to self.filtered_time_frames
+        :param time_frames: the list of time_frame to add
         """
-        cryptofeed.feedhandler.SIGNALS = ()
+        for time_frame in time_frames:
+            self._add_time_frame(time_frame)
 
-    def subscribe_candle_feed(self, exchange_symbols):
-        candle_callback = self.callback_by_feed[self.EXCHANGE_FEEDS[Feeds.CANDLE]]
+    async def reconnect(self):
```

Comment:
I think we should name this method `reset` or something like this but not `reconnect` since it's not really reconnecting the ws